components/src/dynamo/trtllm/tests/test_trtllm_image_diffusion.py

```
@@ -0,0 +1,615 @@
+class TestNvCreateImageRequest:
+    """Tests for NvCreateImageRequest protocol type."""
+
+    def test_required_fields(self):
+        """Test that prompt and model are required."""
+        req = NvCreateImageRequest(prompt="A cat", model="black-forest-labs/FLUX.1-dev")
+        assert req.prompt == "A cat"
+        assert req.model == "black-forest-labs/FLUX.1-dev"
+
+    def test_required_fields_missing_prompt(self):
+        """Test that missing prompt raises validation error."""
+        with pytest.raises(Exception):  # Pydantic ValidationError
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)